### ISSUE TITLE
Dapr gRPC concurrency error

### ISSUE THREAD

**Issue by safari137** [dapr/dapr#4855]
## In what area(s)?

> /area runtime

## What version of Dapr?

>1.7.4

## Expected Behavior

I expect to be able to make concurrent requests to the Dapr side car over gRPC

## Actual Behavior

When making concurrent requests to the Dapr side car over gRPC one request is successful, but all the other concurrent requests get the following .NET exception.

`Grpc.Core.RpcException: Status(StatusCode="Cancelled", Detail="grpc: the client connection is closing")`

This error does not occur if I point the app A to make the gRPC request directly to app B instead of using the Dapr side car.

#### Step 1:
Start application A (client) as follows:
dapr run -a client -P grpc -- dotnet run

Start application B (server) as follows (note: the server has an http2 and http1 port open)
dapr run -a server -P grpc -p 5145 -- dotnet run --project --urls="http://localhost:5145;http://localhost:7145"

#### Step 2:
Make 2-5 concurrent requests from application A -> application B via the dapr side car over gRPC.

#### Step 3:
This may be unimportant, but here is how the gRPC client is setup
```
builder.Services.AddGrpcClient<MyGrpc.MyGrpcClient>(nameof(MyGrpc.MyGrpcClient), options =>
{
    var defaultPort = (Environment.GetEnvironmentVariable("DAPR_GRPC_PORT") ?? "3500");
    var baseURL = (Environment.GetEnvironmentVariable("BASE_URL") ?? "http://localhost");
    baseURL += $":{(Environment.GetEnvironmentVariable("BASE_URL_PORT") ?? defaultPort)}";
    System.Console.WriteLine("gRPC using: " + baseURL);
    options.Address = new Uri(baseURL);
});
```
BASE_URL and BASE_URL_PORT are used to override the app to bypass Dapr.  They are null when this issue occurs, so the value of baseURL is `http://localhost:{DAPR_GRPC_PORT}`

NOTE: I've left out all the code details since this works fine without the Dapr side car.  I can create a small repo if needed though.

## Release Note

RELEASE NOTE:

**Comment by yaron2** [comment]
This is fixed in 1.8 already.

**Comment by safari137** [comment]
After upgrading to 1.8.0-rc4 this works as expected.  Thank you!
